### PR TITLE
Implement mmap and msync for NODERAWFS

### PR DESCRIPTION
**Pull request by kleisauke**
Similar to #10669 but then for NODERAWFS.

### CODE REVIEW

File: src/library_noderawfs.js [hunk 1]
```diff
@@ -122,10 +122,30 @@ mergeInto(LibraryManager.library, {
...
-    mmap: function() {
-      throw new FS.ErrnoError(ERRNO_CODES.ENODEV);
+    mmap: function(stream, address, length, position, prot, flags) {
+      if (stream.stream_ops) {
+        // this stream is created by in-memory filesystem
+        return VFS.mmap(stream, address, length, position, prot, flags);
+      }
+
+      // We don't currently support location hints for the address of the mapping
+      assert(address === 0);
```
Comment (sbc100) [inline src/library_noderawfs.js:132]: Return errno here rather than asserting?

File: tests/fs/test_mmap.c [hunk 2]
```diff
@@ -233,7 +238,7 @@ int main() {
    * is not written beyond the allocated memory area for the mmap operation.
    */
   {
-    int fd = open("/yolo/overallocatedfile.txt", O_RDWR | O_CREAT, (mode_t)0600);
+    int fd = open("./overallocatedfile.txt", O_RDWR | O_CREAT, (mode_t)0600);
```
Comment (sbc100) [inline tests/fs/test_mmap.c:241]: can we instead just replace `/yolo/` with `yolo/` everywhere?    that way the ifdef's around the mkdir won't be needed.

File: tests/test_core.py [hunk 3]
```diff
@@ -5144,12 +5144,14 @@ def test_fs_append(self):
...
     orig_compiler_opts = self.emcc_args[:]
-    for fs in ['MEMFS', 'NODEFS']:
+    for fs in ['MEMFS', 'NODEFS', 'NODERAWFS']:
       src = path_from_root('tests', 'fs', 'test_mmap.c')
       out = path_from_root('tests', 'fs', 'test_mmap.out')
       self.emcc_args = orig_compiler_opts + ['-D' + fs]
       if fs == 'NODEFS':
         self.emcc_args += ['-lnodefs.js']
+      if fs == 'NODERAWFS':
+        self.emcc_args += ['-sNODERAWFS=1']
```
Comment (sbc100) [inline tests/test_core.py:5154]: Does `-lnoderawfs.js` work?  (like the line two lines above?)

File: tests/fs/test_mmap.c (hunk 2 above)
Comment (kleisauke) [reply tests/fs/test_mmap.c]: Done with commit https://github.com/kleisauke/emscripten/commit/244e4786237d284ac3971949fdbfbf97fc904f12, thanks!

File: tests/test_core.py (hunk 3 above)
Comment (kleisauke) [reply tests/test_core.py]: That works too, but in that case also `-lnodefs.js` needs to be linked.

(Perhaps we could just synonym `-sNODERAWFS=1` with `-lnoderawfs.js`? See for e.g. https://github.com/kleisauke/emscripten/commit/b3607b9d591044e00a1be71ad10641b3bb285da3)

File: src/library_noderawfs.js (hunk 1 above)
Comment (kleisauke) [reply src/library_noderawfs.js:132]: Do you know which errno is appropriate in this case? `ENXIO` or just `ENODEV`? Note that we need to change `library_nodefs.js` as well:
https://github.com/emscripten-core/emscripten/blob/097cfb8726001cf81343f1a9f61e06b5074ace85/src/library_nodefs.js#L299-L300

Comment (sbc100) [reply src/library_noderawfs.js:132]: EINVAL seems reasonable enough.

**Review by sbc100 — APPROVED** [review]
lgtm with the errno change.

File: src/library_noderawfs.js (hunk 1 above)
Comment (kleisauke) [reply src/library_noderawfs.js]: Done with commit https://github.com/kleisauke/emscripten/commit/6ac5c096b01875b7e2c10d55a15266e346bacdec, thanks!

File: tests/test_core.py (hunk 3 above)
Comment (kleisauke) [reply tests/test_core.py]: I included `-lnodefs.js` and `-lnoderawfs.js` with commit https://github.com/kleisauke/emscripten/commit/244e4786237d284ac3971949fdbfbf97fc904f12, but perhaps this (i.e. `['-sNODERAWFS=1']` -> `['-lnodefs.js', '-lnoderawfs.js']`) should be done throughout the whole test suite? Let me know if you want a separate PR for this.

**Comment by kleisauke** [comment]
Rebased on top of the master branch to resolve merge conflict.

**Pull request merged**